bonfire/Modules/Tools/Controllers/LogsController.php

```diff
@@ -0,0 +1,259 @@
+              }
+
+            return redirect()->to(ADMIN_AREA.'tools/logs')->with('message', lang('Logs.delete_success'));
+
+          }
+
+        } elseif (isset($_POST['delete_all'])) {
+
+          helper('filesystem');
+
+          if (delete_files($this->logs_path)) {
+
+              // Restore the index.html file.
+              @copy(APPPATH . '/index.html', "{$this->logs_path}index.html");
+              return redirect()->to(ADMIN_AREA.'tools/logs')->with('message', lang('Logs.delete_all_success'));
+
+
+          } else {
+              return redirect()->to(ADMIN_AREA.'tools/logs')->with('error', lang('Logs.delete_error'));
+          }
+      }
+
+        global $app;
+        $db = db_connect();
+
+
+        helper('form');
+
+        $uri = new \CodeIgniter\HTTP\URI();
+
+        $pager = \Config\Services::pager();
+
+        $page = isset($_GET['page']) ? $_GET['page'] : 1;
+
+        // Load the Log Files.
+        $logs = get_filenames($this->logs_path);
+
+        unset($logs[0]);
+
+        $total = count($logs);
+
+        $limit = 15;
+
+        $offset = ($page > 1)? ($page-1)*$limit: 0;
+
+        $pager->makeLinks($page,$limit,$total);
+
+        $logs = array_slice($logs, $offset, $limit);
+
+        return $this->render($this->viewPrefix .'logs', [
+            'logs' => $logs,
+            'pager' => $pager,
+        ]);
+    }
+
+    /**
+     * Show the contents of a single log file.
+     *
+     * @param string $file The full name of the file to view (including extension).
+     *
+     * @return void
+     */
+    public function view(string $file = '')
+    {
+        if (empty($file)) {
+            Template::set_message(lang('logs_view_empty'), 'danger');
+            redirect(SITE_AREA . '/developer/logs');
+        }
+
+        helper('form');
+
+        $pager = \Config\Services::pager();
+
+        $path = $this->logs_path. $file;
+
+        $logs =  $this->processLogs($this->getLogs($path));
+
+        $limit = 15;
+
+        $page = isset($_GET['page']) ? $_GET['page'] : 1;
+
+        $offset = ($page > 1)? ($page-1)*$limit: 0;
+
+        $pager->makeLinks($page,$limit,count($logs));
+
+        $logs = array_slice($logs, $offset, $limit);
+
+        return $this->render($this->viewPrefix .'view', [
+            'pager' => $pager,
+            'log_file' => $file,
+            'canDelete' => 1,
+            'log_content' => $logs,
+            'log_file_pretty' => date('F j, Y', strtotime(str_replace('.log', '', str_replace('log-', '', $file)))),
+        ]);
+
+    }
+
+
+    /*
+    * This function will process the logs. Extract the log level, icon class and other information
+    * from each line of log and then arrange them in another array that is returned to the view for processing
+    *
+    * @params logs. The raw logs as read from the log file
+    * @return array. An [[], [], [] ...] where each element is a processed log line
+    * */
+   private function processLogs($logs) {
```

Comment:
I think these private methods would be better off in a library. This muddies up the controller with stuff that is not controlling the path through the application. This would also give a chance to use Entities for the logs, which could have the methods to extract the different bits. It's a little more abstracted but cleaner, and allows further expansion down the line, if necessary.